docs/hugo/content/design/ADR-2022-11-Change-Detection.md

```diff
@@ -26,24 +28,36 @@ Some fields are optional in the Swagger, but are required by Azure. In these cas
 
 While we can tell the different in a Spec between a field that was never set by the user, and a field that we set to an explicit value (because all our Spec properties are pointers), this doesn't mean we can assume any explicit value in a Status is a default. For example, if a third party makes a change via the Azure portal, we want to detect this as a change and and revert it.
 
+A variation of value defaulting may occur with arrays and maps, where some services will add a default value to the collection if the user does not specify it. Conceptually, you can write `["a"]` and get back `["default", "a"]` when you read it back.
+
 ### Read-After-Write Consistency
 
-Azure is not guaranteed to return the exact same resource that was PUT. For example, the `etag` field may change, or the `id` field may be returned with a different casing. Some resource providers also *normalize* field values, for example the CosmosDB API will return `West US` when the resource specified `westus`.
+Azure is not guaranteed to return the exact same resource that was PUT. 
+
+* Some services normalize identifier fields (converting them to all lowercase)
+* Some fields accept a wider range of values than they return.
 
-Potential mitigation: Generally speaking, Azure is expected to be "case-preserving, case-insensitive", so we should compare all string fields in a case insensitive manner. Region names and virtual machines SKUS are two known special cases where we may need to do some special case handling. For example, regions specified as `eastus` may be returned as `East US`.
+For example the CosmosDB API will return the region `West US` when the resource specified `westus`.
+
+Potential mitigations: 
+
+* Generally speaking, Azure identifiers are expected to be "case-preserving, case-insensitive", so we should compare those string fields in a case insensitive manner. 
+* Region names and virtual machines SKUS are two known special cases where we may need to do some special case handling. 
 
 ### Array ordering
 
-Some resources have arrays of sub-resources. These arrays may not be guaranteed to be returned in the same order as they were specified. 
+Some resources have arrays of items. These arrays may not be guaranteed to be returned in the same order as they were specified. 
 
-Potential mitigation: Where the items in an array have a known identifier (easy for nested resources), use that identifier to match them up. Where the items in an array do not have a known identifier, compare them by index. We may find need for 
+Potential mitigation: In the common case where the items in an array have a known identifier, use that identifier to match them up. Where the items in an array do not have a known identifier, compare them by index.
```

Comment:
```suggestion
Potential mitigation: In the common case where the items in an array have a known identifier (for example: When the items represent sub-resources with an `id` field), use that identifier to match them up. Where the items in an array do not have a known identifier, compare them by index.
```

Reply:
Or possibly a way to phrase it without parens, but which includes that example, which I do think was a good one you had originally highlighted.